src/hooks/useLocationsOfAllTenantsQuery/useLocationsOfAllTenantsQuery.js

```diff
@@ -0,0 +1,52 @@
+import { useQueries } from 'react-query';
+
+import {
+  useNamespace,
+  useOkapiKy,
+  useStripes,
+} from '@folio/stripes/core';
+
+import useConsortiumTenants from '../useConsortiumTenants';
+import { isUserInConsortiumMode } from '../../utils';
+import {
+  CQL_FIND_ALL,
+  LIMIT_MAX,
+  OKAPI_TENANT_HEADER,
+} from '../../constants';
+
+const useLocationsOfAllTenantsQuery = () => {
+  const [namespace] = useNamespace();
+  const stripes = useStripes();
+  const ky = useOkapiKy();
+
+  const { tenants, isLoading } = useConsortiumTenants();
+
+  const tenantIds = tenants.map(tenant => tenant.id);
+
+  const queries = useQueries(tenantIds.map(tenantId => {
+    return {
+      enabled: Boolean(!isLoading && isUserInConsortiumMode(stripes)),
+      queryKey: [namespace, 'tenant', tenantId],
+      queryFn: () => ky.get('locations', {
+        searchParams: {
+          query: CQL_FIND_ALL,
+          limit: LIMIT_MAX,
+        },
+        hooks: {
+          beforeRequest: [
+            request => {
+              request.headers.set(OKAPI_TENANT_HEADER, tenantId);
+            },
+          ],
+        },
+      }).json(),

```

Comment:
just use `useTenantKy` here and you won't need to override the headers

Reply:
We need to use override here because it's inside `map` loop. If we use `useTenantKy` it's going to use the same tenant for all requests, but we need each request with different tenant